Add current and future Chrome and FireFox versions.

paulrutter opened the pull request:
Fix for issue
https://github.com/browscap/browscap/issues/2370 and
https://github.com/browscap/browscap/issues/2356

Based on merged PR's which added future versions of both browsers.
https://github.com/browscap/browscap/pull/2323/files
https://github.com/browscap/browscap/pull/2244/files

paulrutter committed:
Add current and future Chrome and FireFox versions.

Fix for issue
https://github.com/browscap/browscap/issues/2370 and
https://github.com/browscap/browscap/issues/2356

Based on merged PR's which added future versions of both browsers.
https://github.com/browscap/browscap/pull/2323/files
https://github.com/browscap/browscap/pull/2244/files

paulrutter commented:
Hmm, i will probably need to add some code coverage for the new versions.
Will try to look into this later.

paulrutter committed:
Add more future Chrome versions

mimmi20 commented on resources/user-agents/browsers/chrome/chrome-66-on.json:
```diff
@@ -1,6 +1,6 @@
 {
   "division": "Chrome #MAJORVER#.#MINORVER#",
-  "versions": [85, 84, 83, 82, 81, 80, 79, 78, 77, 76, 75, 74, 73, 72, 71, 70, 69, 68, 67, 66],
+  "versions": [92, 91, 90, 89, 88, 87, 86, 85, 84, 83, 82, 81, 80, 79, 78, 77, 76, 75, 74, 73, 72, 71, 70, 69, 68, 67, 66],
```
Please add these versions also to the Chromium and to the Android Version of Chrome

paulrutter replied:
Done!

paulrutter replied:
Hmm will need to add Chromium i guess, one minute

paulrutter replied:
Done!

mimmi20 commented on resources/user-agents/browsers/firefox-on-desktop/firefox-53-on.json:
```diff
@@ -1,6 +1,6 @@
 {
   "division": "Firefox #MAJORVER#.#MINORVER#",
-  "versions": [80, 79, 78, 77, 76, 75, 74, 73, 72, 71, 70, 69, 68, 67, 66, 65, 64, 63, 62, 61, 60, 59, 58, 57, 56, 55, 54, 53],
+  "versions": [88, 87, 86, 85, 84, 83, 82, 81, 80, 79, 78, 77, 76, 75, 74, 73, 72, 71, 70, 69, 68, 67, 66, 65, 64, 63, 62, 61, 60, 59, 58, 57, 56, 55, 54, 53],
```
Please add these versions also to the Android version of Firefox.

paulrutter committed:
Add all FF and Chrome versions to Android as well.

paulrutter committed:
Add current and future Chrome versions to Chromium.

paulrutter committed:
Add current and future Chrome versions to Chrome on iOS and fix headless chrome.

paulrutter committed:
Add current and future Chrome versions to Chrome on Android webviews and Edge (which is based on Chrome nowadays)

paulrutter commented:
I've added the versions to several other places as well (Edge, Android webview, Chrome on iOS, headless Chrome, Chromium, Firefox on Android).

mimmi20 commented:
There was another PR to update the Firefox versions. Please solve the conflicts.

paulrutter committed:
Merge branch '6.0.x' of https://github.com/browscap/browscap into feature/2370-2356-Add-current-and-future-Chrome-and-Firefox-versions

# Conflicts:
#	resources/user-agents/browsers/firefox-on-desktop/firefox-53-on.json

paulrutter commented:
> There was another PR to update the Firefox versions. Please solve the conflicts.

Done! I've added two more future versions to FireFox.

paulrutter commented:
@mimmi20 Should we add the versions to Iron and Silk browsers as well? These seem to be Chrome based as well.

asgrim commented:
@paulrutter that would be super helpful, please! :pray:

paulrutter commented:
> @paulrutter that would be super helpful, please! 🙏

Will do tomorrow morning.

paulrutter committed:
Add current and future Chrome versions to Iron browser and Silk browser, which are both chromium based

paulrutter committed:
Add unit test for detecting Chrome 86

paulrutter commented:
I've added the Chrome versions to both Silk and Iron, although these do not seem to be updated as fast as Chrome itself (latest Iron is now on 84 and 85 (windows). Silk is less clear, although https://developers.whatismybrowser.com/useragents/explore/software_name/silk/ shows that 80 is returned in some cases.
To be safe, i've added versions up until 92 for both of them.
Following https://chromiumdash.appspot.com/schedule, we're good for Chrome until July 2021, after which 93 will be released.

Also, i've added an initial unit test for detecting Chrome 86 on desktop for Windows 10. I don't know how to run these unit tests locally though.

paulrutter committed:
Fix unit test

paulrutter committed:
Add unit test for Chrome 86 on iOS

paulrutter committed:
Fix unit test

paulrutter committed:
Fix unit test

mimmi20 commented:
> I don't know how to run these unit tests locally though.

`vendor/bin/phpunit` on a Linux console.

mimmi20 commented on tests/issues/issue-2370.php:
```diff
@@ -0,0 +1,103 @@
+            'CssVersion' => '3',
+            'Device_Name' => 'iPhone',
+            'Device_Maker' => 'Apple Inc',
+            'Device_Type' => 'Mobile Device',
```
please change to `Mobile Phone`

paulrutter replied:
I had that originally, but it failed the unit test because "Mobile Device" was returned.
Not sure how to fix that.

paulrutter replied:
I left out that test for "standard", and now "Mobile Phone" passes for both lite and full.
Not sure why "standard" gives Mobile Device instead of Mobile Phone though.

paulrutter committed:
Fix unit test

paulrutter commented:
@mimmi20 @asgrim Thanks for your quick response and all your work for maintaining this project!
It's our main source of truth when categorizing user-agents, which is used in https://github.com/blueconic/browscap-java.

paulrutter commented:
@asgrim i saw the automatic release process (nice work!), does this also update the artifacts on www.browscap.org? Or is that a manual process still?

asgrim commented:
@paulrutter The theory is that Dependabot should automatically notice the release of `browscap/browscap` and send a PR to `browscap/browscap-site`, which I click merge, and then Heroku notices the merge and automatically deploys the site.... This is the first time trying it, so I'll keep an eye on it. The biggest factor will be how long it takes Dependabot to send the PR I suppose, I'm not sure how to affect that currently.

asgrim commented:
@paulrutter I've discovered I can manually trigger Dependabot to check for updates. I suppose eventually that can be automated, but I've updated our "release procedure" docs so next time it will be fairly quick :) 

6.0.41 is now available on browscap.org :+1:

paulrutter commented:
Nice, thanks!
